Code/Framework/AzCore/AzCore/Task/TaskExecutor.cpp

```diff
@@ -25,6 +25,16 @@ namespace AZ
 {
     namespace Internal
     {
+        static constexpr uint32_t NumTrackedRecentDeallocations = 32u;
+        struct DeallocationData
+        {
+            const char* m_parentLabel = nullptr;
+            CompiledTaskGraph* m_ctg = nullptr;
+        };
+        static DeallocationData s_recentDeallocations[NumTrackedRecentDeallocations]; // deallocation breadcrumbs to help look for a double delete
+        static uint32_t s_nextDeallocationSlot = 0;
+        static AZStd::mutex s_deallocationMutex;
```

Comment:
should this be wrapped in `#ifndef RELEASE` or some kind of #ifdef that can be used to disable the lock/deallocation checking?

Reply:
This mutex only protects against the CompiledTaskGraph destruction in a single module memory space(dll).

Because AzCore is a static library that links into multiple links multiple dlls

Reply:
are you saying the performance cost is negligible?

Reply:
We usually only create/free 10's of CompiledTaskGraph's (CTG)per frame, At the moment I'm not expecting this to be a performance bottleneck. Additionally this is short term debug code so we can get extra info about previously deleted CTG's when people hit it randomly. My expectation is that this code would be removed once that bug has been solved.
A comment to that effect is probably warranted